### PR TITLE
For #11015: remove unnecessary BufferedReader in Response.Body.string()

### PR DESCRIPTION
**Pull request by mcomella**
I'm not sure if this needs a changelog entry.

Here's the allocation before (see the rightmost column for "shallow size changed"):
Screen Shot 2021-09-27 at 15 55 51

And after:
Screen Shot 2021-09-27 at 15 44 31

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/concept/fetch/src/test/java/mozilla/components/concept/fetch/ResponseTest.kt [hunk 1]
```diff
@@ -210,7 +211,7 @@ class ResponseTest {
 
         body.string()
 
-        verify(stream).close()
+        verify(stream, atLeastOnce()).close()
```
Comment (pocmo) [inline components/concept/fetch/src/test/java/mozilla/components/concept/fetch/ResponseTest.kt:214]: I guess this comes from the double usage of `use`? I guess we could get rid of the `useStream()` block now? 🤔

**Review by pocmo — APPROVED** [review]

File: components/concept/fetch/src/test/java/mozilla/components/concept/fetch/ResponseTest.kt (hunk 1 above)
Comment (mcomella) [reply components/concept/fetch/src/test/java/mozilla/components/concept/fetch/ResponseTest.kt:214]: Yes. In my head, I knew we only needed one but it irrationally felt dangerous to write. 😅 We have the test to cover that case though so it should be safe.

> I guess we could get rid of the useStream() block now?

Ah, I see. With these two changes in mind, I could change it to:
```kotlin
        fun string(charset: Charset? = null): String = use {
            // <that big comment>
            stream.reader(charset ?: this.charset).readText()
        }
```

I'll update the code to do this and remove the changes from the test.

Comment (mcomella) [reply components/concept/fetch/src/test/java/mozilla/components/concept/fetch/ResponseTest.kt]: @pocmo Can you confirm this is the change you were expecting and, if so, flag this to land?

Comment (pocmo) [reply components/concept/fetch/src/test/java/mozilla/components/concept/fetch/ResponseTest.kt]: Looks good 👍

**Pull request merged**